pkg/controller/keycloak/keycloak_migrations.go

```
@@ -58,3 +74,35 @@ func findDeployment(desiredState *common.DesiredClusterState) *v13.StatefulSet {
 	}
 	return nil
 }
+
+func oneTimeLocalBackup(cr *v1alpha1.Keycloak, currentState *common.ClusterState, desiredState common.DesiredClusterState) (common.DesiredClusterState, error) {
+	keycloakBackup := currentState.KeycloakBackup
+	switch {
+	case keycloakBackup == nil:
+		backupCr := &v1alpha1.KeycloakBackup{}
+		backupCr.Namespace = cr.Namespace
+		backupCr.Name = model.MigrateBackupName + "-" + common.BackupTime
+		labelSelect := metav1.LabelSelector{
+			MatchLabels: cr.Labels,
+		}
+		backupCr.Spec.InstanceSelector = &labelSelect
+
+		migrationBackupCR := common.GenericCreateAction{
+			Ref: model.KeycloakMigrationOneTimeBackup(backupCr),
+			Msg: "Create Local Backup CR",
+		}
+
+		backupDesiredState := common.DesiredClusterState{}
+		backupDesiredState = backupDesiredState.AddAction(migrationBackupCR)
+		return backupDesiredState, nil
+	case keycloakBackup.Status.Phase == v1alpha1.BackupPhaseCreated:
+		log.Info("migrate backup succeeds")
+		return desiredState, nil
+	case keycloakBackup.Status.Phase == v1alpha1.BackupPhaseFailing:
+		backUpFail := errors.New("migrate backup fails")
+		return nil, backUpFail
+	default:
+		emptyDesiredState := common.DesiredClusterState{}
```

We should probably report an error here. This is an unexpected situation.